docs/tuning-for-production.md

```diff
@@ -6,6 +6,20 @@ title: Tuning the ProxyManager for production
 
 By default, all proxy factories generate the required proxy classes at runtime.
 
+To generate proxies and store them as files, you need to use `FileWriterGeneratorStrategy` by passing it to the configuration before setting up other factories:
+
+```php
+    
+$config = new \ProxyManager\Configuration();
+
+$fileLocator = new \ProxyManager\FileLocator\FileLocator(__DIR__.'/my/generated/classes/cache/dir');
+$config->setGeneratorStrategy(new \ProxyManager\GeneratorStrategy\FileWriterGeneratorStrategy($fileLocator));
+
+// pass the configuration to proxymanager factory
+$factory = new ProxyManager\Factory\LazyLoadingValueHolderFactory($config);
```

Comment:
We may need a full example with this combined with the autoloader.

Reply:
Is `$config->setProxiesTargetDir()` absolutely necessary for generating the proxies? My implementation generates and uses the proxies from Auryn, so it needs that line otherwise it will throw a `ReflectionException`

Reply:
The two examples can be put together as one. Is that an option?

Reply:
@2Dsharp yes, making a single example is a good idea.

As for `$config->setProxiesTargetDir()`, it is not strictly needed, but you cannot rely on `ProxyManager\Configuration` to build correct instances without it...

For example, the `FileLocator` is already correctly configured if `setProxiesTargetDir()` was used.

Reply:
I am merging the two examples into one.